soda/core/tests/data_source/test_check_identity.py

```diff
@@ -141,3 +144,31 @@ def test_explicitely_specified_check_identity(scanner: Scanner):
 
     # check that the identity remains the same after changing the check (threshold in this case)
     assert scan_result["checks"][0]["identity"] == row_count_identity
+
+
+def test_for_each_identity(scanner: Scanner):
+    """Tests that same check generated by "for each" clause and manually will have unique identity and definition."""
+    customers_table_name = scanner.ensure_test_table(customers_test_table)
+    customers_dist_table_name = scanner.ensure_test_table(customers_dist_check_test_table)
+
+    scan_result = execute_scan_and_get_scan_result(
+        scanner,
+        f"""
+          for each table T:
+            tables:
+                - {customers_table_name}
+                - {customers_dist_table_name}
+            checks:
+            - row_count > 0
+          checks for {customers_table_name}:
+            - row_count > 0
+        """,
+    )
+    assert scan_result["checks"][0]["identity"] == "4bb40424"
+    assert f"checks for {customers_table_name}" in scan_result["checks"][0]["definition"]
+
+    assert scan_result["checks"][1]["identity"] == "e8dd2881"
```

Comment:
Are you sure this is going to be consistent across invocations? Perhaps it is just better to check that the identities are different and we don't really need to check the identity value itself.